integrationTests/consensus/consensusSigning_test.go

```diff
@@ -10,88 +10,95 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-// TODO: refactor to use nodes from multiple shards
 func initNodesWithTestSigner(
+	numMetaNodes,
 	numNodes,
 	consensusSize,
 	numInvalid uint32,
 	roundTime uint64,
 	consensusType string,
-) []*integrationTests.TestConsensusNode {
+) map[uint32][]*integrationTests.TestConsensusNode {
 
 	fmt.Println("Step 1. Setup nodes...")
 
 	nodes := integrationTests.CreateNodesWithTestConsensusNode(
-		1,
+		int(numMetaNodes),
 		int(numNodes),
 		int(consensusSize),
 		roundTime,
 		consensusType,
 	)
 
-	for _, nodesList := range nodes {
-		displayAndStartNodes(nodesList)
+	for shardID, nodesList := range nodes {
+		displayAndStartNodes(shardID, nodesList)
 	}
 
 	time.Sleep(p2pBootstrapDelay)
 
-	if numInvalid < numNodes {
-		for i := uint32(0); i < numInvalid; i++ {
-			iCopy := i
-			nodes[0][i].MultiSigner.CreateSignatureShareCalled = func(privateKeyBytes, message []byte) ([]byte, error) {
-				fmt.Println("invalid sig share from ",
-					getPkEncoded(nodes[0][iCopy].NodeKeys.Pk),
-				)
-				return []byte("invalid sig share"), nil
+	for shardID := range nodes {
+		if numInvalid < numNodes {
+			for i := uint32(0); i < numInvalid; i++ {
+				iCopy := i
+				nodes[shardID][i].MultiSigner.CreateSignatureShareCalled = func(privateKeyBytes, message []byte) ([]byte, error) {
+					fmt.Println("invalid sig share from ",
+						getPkEncoded(nodes[shardID][iCopy].NodeKeys.Pk),
+					)
+					return []byte("invalid sig share"), nil
+				}
 			}
 		}
 	}
 
-	return nodes[0]
+	return nodes
 }
 
 func TestConsensusWithInvalidSigners(t *testing.T) {
 	if testing.Short() {
 		t.Skip("this is not a short test")
 	}
 
+	numMetaNodes := uint32(4)
 	numNodes := uint32(4)
 	consensusSize := uint32(4)
-	numInvalid := uint32(1)
+	numInvalid := uint32(0)
```

Comment:
i set it to 0 since this test is more relevant when next PR with pseudo slashing will come in, but I checked more and there was an issue there, thanks